### PR TITLE
Option: Information Refresh Time https://tools.ietf.org/html/rfc8415

### PR DESCRIPTION
**Pull request by mcphailtom**
Added structure, parsing and convenience fuctions for OptInformationRefreshTime (32) https://tools.ietf.org/html/rfc8415

### CODE REVIEW

File: dhcpv6/option_informationrefreshtime.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package dhcpv6
+
+import (
+	"fmt"
+	"github.com/u-root/u-root/pkg/uio"
```
Comment (pmazzini) [inline dhcpv6/option_informationrefreshtime.go:5]: NIT: use a separate blocks for external dependency

https://github.com/golang/go/wiki/CodeReviewComments#imports

File: dhcpv6/dhcpv6message.go [hunk 2]
```diff
@@ -248,6 +248,21 @@ func (mo MessageOptions) ElapsedTime() time.Duration {
 	return 0
 }
 
+// InformationRefreshTime returns the Information Refresh Time option
+// as defined by RFC 815 Section 21.23.
+//
+// ElapsedTime returns a duration of 0 if the option is not present.
```
Comment (Natolumin) [inline dhcpv6/dhcpv6message.go:254]: This makes it impossible to differentiate packets with the explicit "0" value (that's not valid per the rfc, but it's still representable on the wire so we should be able to handle it) from packets where the option isn't supplied.

Can you make it either a pointer or return `(Duration, error)` or `(Duration, bool)` where the boolean indicates if it was found, similar to the interface of maps ?

Additionally, there's a copy-paster error, `ElapsedTime` -> `InformationRefreshTime`

File: dhcpv6/option_informationrefreshtime.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	return fmt.Sprintf("InformationRefreshTime: %v", op.InformationRefreshtime)
+}
+
+// parseOptInformationRefreshTime builds an optInformationRefreshTime structure from a sequence
+// of bytes. The input data does not include option code and length bytes.
+func parseOptInformationRefreshTime(data []byte) (*optInformationRefreshTime, error) {
+	var opt optInformationRefreshTime
+	buf := uio.NewBigEndianBuffer(data)
+
+	var irt Duration
```
Comment (Natolumin) [inline dhcpv6/option_informationrefreshtime.go:42]: offtopic note: Duration doesn't handle the special value of `0xffffffff` to mean infinity, which we should fix (but not in this PR)

File: dhcpv6/option_informationrefreshtime.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
+package dhcpv6
+
+import (
+	"fmt"
+	"github.com/u-root/u-root/pkg/uio"
+	"time"
+)
+// OptInformationRefreshTime implements OptionInformationRefreshTime option.
+// https://tools.ietf.org/html/rfc8415
```
Comment (Natolumin) [inline dhcpv6/option_informationrefreshtime.go:9]: nit: for a bit more accuracy
```suggestion
// https://tools.ietf.org/html/rfc8415#section-21.23
```

File: dhcpv6/dhcpv6message.go (hunk 2 above)
Comment (mcphailtom) [reply dhcpv6/dhcpv6message.go]: Great idea, apologies I got a bit lazy here (evidenced by the copy paste error) :)

Comment (pmazzini) [reply dhcpv6/dhcpv6message.go]: I wonder if we should make it consistent with the v4 one in this case:

```
func (d *DHCPv4) IPAddressRenewalTime(def time.Duration) time.Duration {
```

https://github.com/insomniacslk/dhcp/blob/master/dhcpv4/dhcpv4.go#L690

I am not sure what the best option is, just asking for opinion.

Comment (Natolumin) [reply dhcpv6/dhcpv6message.go]: I don't think there is a lot of value in trying to keep the interfaces the same; the protocols are different and I still haven't seen any code that really manages to handle both transparently. The current APIs are already very different

So the question is, do we have users for this api, and what do they care about ?
 * They want to get a (valid) value, anything: then the version with a default is good
 * They want to know if the option was passed, and if yes what the value is : then we want the double-return

As far as I can tell, there are no users on github of either [IPAddressRenewalTime](https://github.com/search?q=IPAddressRenewalTime&type=Code)(no hits except this lib) or [ElapsedTime](https://github.com/search?l=Go&p=3&q=dhcpv6+ElapsedTime&type=Code)(None of the hits are users of this lib). 
So maybe the simplest solution here is to not have a getter at all, until we have a client that wants it and knows what they want ?

**Comment by mcphailtom** [comment]
So I'm happy to provide a single data point for consumption... I consume (privately). Personally I don't really mind which way it goes, though the default option basically just shortens the path to what I would ultimately do as a consumer anyway.

**Comment by pmazzini** [comment]
I'll merge it tomorrow if there are no other comments.

**Pull request merged**